app/facades/food_facade.rb

```diff
@@ -0,0 +1,15 @@
+# app/facades/food_facade.rb
+
+class FoodFacade
+  def self.total_hits(query)
+    json = FoodService.foods_containing_ingredient(query)
+    @total_hits = json[:totalHits].to_i
+  end
+
+  def self.foods_containing_ingredient(query)
+    json = FoodService.foods_containing_ingredient(query)
+    @foods = json[:foods].map do |food|
+      Food.new(food)
+    end
+  end
```

Comment:
Since `total_hits` is calling this method, we are hitting the same api twice. Instead we can do something like this:

```ruby
  def self.foods_containing_ingredient(query)
    json = FoodService.foods_containing_ingredient(query)
    foods = json[:foods].map do |food|
      Food.new(food)
    end
    [json[:totalHits], foods]
  end
  ```
  
 And then in the controller, assign both vars at once and skip the second API call altogether:
 `@total_hits, @foods = FoodFacade.foods_containing_ingredient(params[:q])`